### PR TITLE
ensure that TryParse is used for all user input in editorconfig

### PR DESCRIPTION
**Pull request by jmarolf**
We got several reports in RC of Parse(int) throwing exceptions.  This PR changes the  editorconfig parsing of options to only use TryParse.  It also simplifies several cases where we were passing a parsing function unnecessarily.

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide please review

File: src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs [hunk 1]
```diff
@@ -27,6 +27,10 @@ public bool TryParseReadonlyDictionary(IReadOnlyDictionary<string, object> allRa
                 if (allRawConventions.TryGetValue(KeyName, out object value))
                 {
                     result = _parseValue(value.ToString(), type);
+                    if (result == null)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: return result != null;

File: src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs [hunk 2]
```diff
@@ -49,11 +53,19 @@ public EditorConfigStorageLocation(string keyName)
             {
                 if (type == typeof(int))
                 {
-                    return int.Parse(s);
+                    if (int.TryParse(s, out var value))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: return int.TryParse(...) ? value : null;

File: src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs [hunk 3]
```diff
@@ -49,11 +53,19 @@ public EditorConfigStorageLocation(string keyName)
...
+                    {
+                        return value;
+                    }
+                    return null;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: what does null mean for an int?  should this be 0?

File: src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs [hunk 4]
```diff
@@ -49,11 +53,19 @@ public EditorConfigStorageLocation(string keyName)
...
                 }
                 else if (type == typeof(bool))
                 {
-                    return bool.Parse(s);
+                    if (bool.TryParse(s, out var value))
+                    {
+                        return value;
+                    }
+                    return null;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: should this be false?  or does 'null' just mean 'use the default'?

Comment (jmarolf) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: null in this case means use the default value.

---
In reply to: [102832596](https://github.com/dotnet/roslyn/pull/17097#discussion_r102832596) [](ancestors = 102832596)

File: src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs (hunk 3 above)
Comment (jmarolf) [inline src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs]: null means we couldn't parse it so return the default value.

---
In reply to: [102832529](https://github.com/dotnet/roslyn/pull/17097#discussion_r102832529) [](ancestors = 102832529)

**Comment by jmarolf** [comment]
retest windows_release_unit64_prtest please

File: src/Workspaces/CSharp/Portable/CodeStyle/CSharpCodeStyleOptions.cs [hunk 5]
```diff
@@ -12,79 +12,79 @@ internal static class CSharpCodeStyleOptions
         public static readonly Option<CodeStyleOption<bool>> UseImplicitTypeForIntrinsicTypes = new Option<CodeStyleOption<bool>>(
             nameof(CodeStyleOptions), nameof(UseImplicitTypeForIntrinsicTypes), defaultValue: CodeStyleOption<bool>.Default,
             storageLocations: new OptionStorageLocation[] {
-                new EditorConfigStorageLocation("csharp_style_var_for_built_in_types", ParseEditorConfigCodeStyleOption),
+                new EditorConfigStorageLocation("csharp_style_var_for_built_in_types"),
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/CodeStyle/CSharpCodeStyleOptions.cs]: Move `ParseEditorConfigCodeStyleOption` closer to `EditorConfigStorageLocation`.

Also consider caching the delegate for the parse function used in the 1-arg version of the `EditorConfigStorageLocation` constructor

**Pull request merged**